intelmq/lib/utils.py

```diff
@@ -808,3 +814,46 @@ def file_name_from_response(response: requests.Response) -> str:
     except KeyError:
         file_name = response.url.split("/")[-1]
     return file_name
+
+
+def list_all_bots() -> str:
+    bot_groups = ['Collectors', 'Parsers', 'Experts', 'Outputs']
+    bot_types = [
+        'CollectorBot',
+        'ParserBot',
+        'OutputBot',
+        'Bot'
+    ]
+    bots = {}
+    for bot_group in bot_groups:
+        bots[bot_group] = {}
+
+    botfiles = [botfile for botfile in pathlib.Path('intelmq/bots').glob('**/*.py') if botfile.is_file() and botfile.name != '__init__.py']
+    for file in botfiles:
+        mod = importlib.import_module("intelmq.bots.{}.{}.{}".format(file.parts[-3:][0], file.parts[-2:][0], file.stem))
+        if hasattr(mod, 'BOT'):
+            name = mod.BOT.__name__
+            keys = {}
+            variables = sorted(
+                (name, value) for name, value in vars(mod.BOT).items()
+                if not inspect.ismethod(value) and
+                not inspect.isfunction(value) and
+                not inspect.isclass(value) and
+                not inspect.isroutine(value) and
+                not name.isupper() and
+                not name.startswith('_')
+            )
+            for key, value in variables:
+                keys[key] = value
+
+            for t in bot_types:
+                if name == t:
+                    continue
```

Comment:
I don't think that is needed anymore, using the code above bots whose name are the same as one of the `bot_types` are not listed. Also `t` is not a meaningful variable name. Given that `bot_types` is not needed anywhere else in the method, you can also simpy write
```
# Some bots have their parent class names as part of their name,
# we are removing that to make the name more readable.
for bot_type in ['CollectorBot','ParserBot','OutputBot','Bot']:
    name = name.replace(bot_type, '')
```